File: repos_estudiantes_ALVAREZ_segunda_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/HERENCIA/EJERCICIO1/JAVA-EJERCICIO1/src/Ejercicio1/Ej1.java
package Ejercicio1;

public class Ej1 {
    public static void main(String[] args) {
        Coche coche1 = new Coche("Toyota", "Corolla", 2025, 20000, 4, "Gasolina");
        Coche coche2 = new Coche("Ford", "Mondeo", 2025, 25000, 5, "Diésel");
        Coche coche3 = new Coche("Honda", "Civic", 2024, 22000, 2, "Híbrido");

        Moto moto1 = new Moto("Yamaha", "MT-07", 2025, 9000, 689, "4T");
        Moto moto2 = new Moto("Suzuki", "GSX-R600", 2023, 11000, 599, "4T");

        System.out.println("-------Información de coches:-------");
        coche1.mostrar_info();
        System.out.println();
        coche2.mostrar_info();
        System.out.println();
        coche3.mostrar_info();

        System.out.println("-------Información de motos:-------");
        moto1.mostrar_info();
        System.out.println();
        moto2.mostrar_info();

        System.out.println("-------Coches con más de 4 puertas:-------");
        Coche[] coches = {coche1, coche2, coche3};
        for (Coche coche : coches) {
            if (coche.get_num_puertas() > 4) {
                coche.mostrar_info();
                System.out.println();
            }
        }

        System.out.println("-------Vehículos gestión 2025:-------");
        Vehiculo[] vehiculos = {coche1, coche2, coche3, moto1, moto2};
        for (Vehiculo v : vehiculos) {
            if (v.get_año() == 2025) {
                v.mostrar_info();
                System.out.println();
            }
        }
    }
}

class Vehiculo {
    private String marca;
    private String modelo;
    private int año;
    private double precio_base;

    public Vehiculo(String marca, String modelo, int año, double precio_base) {
        this.marca = marca;
        this.modelo = modelo;
        this.año = año;
        this.precio_base = precio_base;
    }

    public String get_marca() {
        return marca;
    }

    public void set_marca(String marca) {
        this.marca = marca;
    }

    public String get_modelo() {
        return modelo;
    }

    public void set_modelo(String modelo) {
        this.modelo = modelo;
    }

    public int get_año() {
        return año;
    }

    public void set_año(int año) {
        this.año = año;
    }

    public double get_precio_base() {
        return precio_base;
    }

    public void set_precio_base(double precio_base) {
        this.precio_base = precio_base;
    }

    public void mostrar_info() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Año: " + año + ", Precio Base: " + precio_base);
    }
}

class Coche extends Vehiculo {
    private int num_puertas;
    private String tipo_combustible;

    public Coche(String marca, String modelo, int año, double precio_base, int num_puertas, String tipo_combustible){
        super(marca, modelo, año, precio_base);
        this.num_puertas = num_puertas;
        this.tipo_combustible = tipo_combustible;
    }

    public int get_num_puertas() {
        return num_puertas;
    }

    public void set_num_puertas(int num_puertas) {
        this.num_puertas = num_puertas;
    }

    public String get_tipo_combustible() {
        return tipo_combustible;
    }

    public void set_tipo_combustible(String tipo_combustible) {
        this.tipo_combustible = tipo_combustible;
    }

    @Override
    public void mostrar_info() {
        super.mostrar_info();
        System.out.println("Número de puertas: " + num_puertas + ", Tipo de combustible: " + tipo_combustible);
    }
}

class Moto extends Vehiculo {
    private int cilindrada;
    private String tipo_motor;

    public Moto(String marca, String modelo, int año, double precio_base, int cilindrada, String tipo_motor) {
        super(marca, modelo, año, precio_base);
        this.cilindrada = cilindrada;
        this.tipo_motor = tipo_motor;
    }

    public int get_cilindrada() {
        return cilindrada;
    }

    public void set_cilindrada(int cilindrada) {
        this.cilindrada = cilindrada;
    }

    public String get_tipo_motor() {
        return tipo_motor;
    }

    public void setTipoMotor(String tipo_motor) {
        this.tipo_motor = tipo_motor;
    }

    @Override
    public void mostrar_info() {
        super.mostrar_info();
        System.out.println("Cilindrada: " + cilindrada + ", Tipo de motor: " + tipo_motor);
    }
}

File: repos_estudiantes_ALVAREZ_segunda_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/HERENCIA/EJERCICIO1/Ej1.py
class Vehiculo:
    def __init__(self, marca, modelo, año, precio_base):
        self.__marca = marca
        self.__modelo = modelo
        self.__año = año
        self.__precio_base = precio_base

    def get_marca(self):
        return self.__marca

    def get_modelo(self):
        return self.__modelo

    def get_año(self):
        return self.__año

    def get_precio_base(self):
        return self.__precio_base

    def set_marca(self, marca):
        self.__marca = marca

    def set_modelo(self, modelo):
        self.__modelo = modelo

    def set_año(self, año):
        self.__año = año

    def set_precio_base(self, precio_base):
        self.__precio_base = precio_base

    def mostrar_info(self):
        print(f"Marca: {self.__marca}, Modelo: {self.__modelo}, Año: {self.__año}, Precio Base: {self.__precio_base}")


class Coche(Vehiculo):
    def __init__(self, marca, modelo, año, precio_base, num_puertas, tipo_combustible):
        super().__init__(marca, modelo, año, precio_base)
        self.__num_puertas = num_puertas
        self.__tipo_combustible = tipo_combustible

    def get_num_puertas(self):
        return self.__num_puertas

    def get_tipo_combustible(self):
        return self.__tipo_combustible

    def set_num_puertas(self, num_puertas):
        self.__num_puertas = num_puertas

    def set_tipo_combustible(self, tipo_combustible):
        self.__tipo_combustible = tipo_combustible

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Número de puertas: {self.__num_puertas}, Tipo de combustible: {self.__tipo_combustible}")


class Moto(Vehiculo):
    def __init__(self, marca, modelo, año, precio_base, cilindrada, tipo_motor):
        super().__init__(marca, modelo, año, precio_base)
        self.__cilindrada = cilindrada
        self.__tipo_motor = tipo_motor

    def get_cilindrada(self):
        return self.__cilindrada

    def get_tipo_motor(self):
        return self.__tipo_motor

    def set_cilindrada(self, cilindrada):
        self.__cilindrada = cilindrada

    def set_tipo_motor(self, tipo_motor):
        self.__tipo_motor = tipo_motor

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Cilindrada: {self.__cilindrada}, Tipo de motor: {self.__tipo_motor}")

coche1 = Coche("Toyota", "Corolla", 2025, 20000, 4, "Gasolina")
coche2 = Coche("Ford", "Mondeo", 2025, 25000, 5, "Diésel")
coche3 = Coche("Honda", "Civic", 2024, 22000, 2, "Híbrido")
moto1 = Moto("Yamaha", "MT-07", 2025, 9000, 689, "4T")
moto2 = Moto("Suzuki", "GSX-R600", 2023, 11000, 599, "4T")

print("-------Información de coches:-------")
coche1.mostrar_info()
print()
coche2.mostrar_info()
print()
coche3.mostrar_info()
print("-------Información de motos:-------")
moto1.mostrar_info()
print()
moto2.mostrar_info()

print("-------Coches con más de 4 puertas:-------")
coches = [coche1, coche2, coche3]
for coche in coches:
    if coche.get_num_puertas() > 4:
        coche.mostrar_info()
        print()

print("-------Vehículos gestión 2025:-------")
vehiculos = coches + [moto1, moto2]
for v in vehiculos:
    if v.get_año() == 2025:
        v.mostrar_info()
        print()